src/cpp/session/modules/SessionCodeSearch.cpp

```diff
@@ -69,18 +69,27 @@ namespace code_search {
 
 namespace {
 
-bool isInCmakeBuildDirectory(const FilePath& filePath)
+bool isWithinIgnoredDirectory(const FilePath& filePath)
 {
-   FilePath parentPath = filePath.parent();
-   while (!parentPath.empty()
-          && parentPath != projects::projectContext().directory())
+   for (FilePath parentPath = filePath.parent();
+        !parentPath.empty();
+        parentPath = parentPath.parent())
```

Comment:
1. Good question -- I was following the Boost documentation which seems to indicate that attempting to get the parent path of a root directory would return an empty path, but I'm not sure if that's true for root directories on Windows.

2. I was thinking that this should work in case we decided to monitor files outside of the current active project, but since we aren't likely to do this any time soon (at least, for the code search indexer) I should change that back.

Thanks!